File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_dict = defaultdict(int)

        for i in range(len(nums)):
            num = nums[i]
            if target - num in num_dict:
                return [num_dict[target - num], i]
            
            num_dict[num] = i

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> num_map = new HashMap<>();
        
        int[] res = new int[2];
        
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int target_num = target - num;

            if (num_map.containsKey(target_num)) {
                res[0] = num_map.get(target_num);
                res[1] = i;
                break;
            }

            num_map.put(num, i);
        }

        return res;
    }
}